synapse/rest/client/room.py

```diff
@@ -1140,14 +1140,10 @@ async def on_POST(
 class RoomHierarchyRestServlet(RestServlet):
     PATTERNS = [
         re.compile(
-            "^/_matrix/client/unstable/org.matrix.msc2946"
+            "^/_matrix/client/(v1|unstable/org.matrix.msc2946)"
```

Comment:
Paranoid: do we refer to capture groups by their index anywhere? The `P<room_id>` makes it look like we always use named capture groups so probably not.

As a sanity check, I tried grepping for `\|unstable`, but didn't find any matches outside of `configure_workers_and_start`. I guess this is the CLIENT_PATTERNS stuff discussed here and elsewhere. (I don't have an opinion on that and am happy to go with the consensus.)

Reply:
Good question! I wasn't entirely sure, but we end up only passing on [`groupdict()`](https://docs.python.org/3/library/re.html#re.Match.groupdict) from the resulting `Match` object, which does only pass *named* subgroups.

See https://github.com/matrix-org/synapse/blob/b64b6d12d47c012a9996a2585a969d3b962d25fa/synapse/http/server.py#L419-L425

Reply:
(I can change it to the suggestion of two different URL paths if you'd prefer, which seems to be what we do in other places. I think there's a benefit to having it match the docs too though!)

Reply:
> (I can change it to the suggestion of two different URL paths if you'd prefer, which seems to be what we do in other places. I think there's a benefit to having it match the docs too though!)

I think it's fine as it is, and matching docs is a nice-to-have!